src/post/post.controller.ts

```diff
@@ -0,0 +1,72 @@
+import {
+  Body,
+  Controller,
+  Delete,
+  Get,
+  Param,
+  Patch,
+  Post,
+  Req,
+  UseGuards,
+} from '@nestjs/common'
+import { AuthGuard } from '@nestjs/passport'
+import { Request } from 'express'
+import { ApiCreatedResponse, ApiOkResponse } from '@nestjs/swagger'
+import PostDto from './dto/createPost.dto'
+import PostService from './post.service'
+import UpdatePostDto from './dto/updatePost.dto'
+import CreatedPostDto from './dto/CreatedPost.dto'
+
+interface MyUserRequest extends Request {
```

Comment:
Try to refactor this to a User decorator and use an authentication guard to assign request user. This would be a convenient way to access request user's data on all our authenticated requests.